halo2_gadgets/src/ecc/chip/mul_fixed.rs

```diff
@@ -467,40 +447,54 @@ impl ScalarFixed {
                     let word = z_cur
                         .zip(z_next)
                         .map(|(z_cur, z_next)| z_cur - z_next * *H_BASE);
+                    // This assumes that the endianness of the encodings of pallas::Base
+                    // and pallas::Scalar are the same. They happen to be, but we need to
+                    // be careful if this is generalised.
```

Comment:
This also comes up in recursion when we reduce some challenges from one field to another. Perhaps we should have a utility function or trait for reducing an element in one field modulo the other.

Reply:
Yeah, we have those in `orchard`. I figured I'd just leave a comment while I was here.